presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import io.airlift.slice.XxHash64;
+
+import static java.lang.Long.toHexString;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class QueryInfoUtils
+{
+    private QueryInfoUtils() {}
+
+    public static String computeQueryHash(String query)
```

Comment:
Nit: please add a preconditon.

`requireNonNull(query);`

Reply:
Updated to add the precondition :).